src/fs/read_dir.rs

```diff
@@ -80,7 +80,7 @@ enum State {
 
 impl ReadDir {
     /// Creates an asynchronous `ReadDir` from a synchronous handle.
-    pub(crate) fn new(inner: std::fs::ReadDir) -> ReadDir {
+    pub(crate) const fn new(inner: std::fs::ReadDir) -> ReadDir {
```

Comment:
Can this even _be_ const? Even `inner` will be different, how would it generate code to account for that?

Reply:
I'm not sure I understand your reasoning here